### ISSUE TITLE
1.19.0: DOMElement *kit(WebCore::Element *) was called from a secondary thread

### ISSUE THREAD

**Issue by Kapeli** [sparkle-project/Sparkle#1256]
I'm getting this crash with 1.19.0:

```
Application Specific Information:
*** Terminating app due to uncaught exception 'WebKitThreadingException', reason: 'DOMElement *kit(WebCore::Element *) was called from a secondary thread'

Last Exception Backtrace:
0   CoreFoundation                       0x00007fff2b434313 __exceptionPreprocess + 147
1   libobjc.A.dylib                      0x00007fff525a2c76 objc_exception_throw + 48
2   CoreFoundation                       0x00007fff2b4c5dcd +[NSException raise:format:] + 205
3   WebKitLegacy                         0x00007fff39e033c7 kit(WebCore::Element*) + 22
4   WebKitLegacy                         0x00007fff39e6a73a WebChromeClient::elementDidFocus(WebCore::Element&) + 25
5   WebCore                              0x00007fff3936588d WebCore::Element::dispatchFocusEvent(WTF::RefPtr<WebCore::Element, WTF::DumbPtrTraits<WebCore::Element> >&&, WebCore::FocusDirection) + 60
6   WebCore                              0x00007fff387484f9 WebCore::dispatchEventsOnWindowAndFocusedElement(WebCore::Document*, bool) + 312
7   WebCore                              0x00007fff3967b774 WebCore::FocusController::setActivityState(unsigned int) + 67
8   WebCore                              0x00007fff396969de WebCore::Page::setActivityState(unsigned int) + 61
9   CoreFoundation                       0x00007fff2b3c3f2c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
10  CoreFoundation                       0x00007fff2b3c3dfa _CFXRegistrationPost + 458
11  CoreFoundation                       0x00007fff2b3c3b31 ___CFXNotificationPost_block_invoke + 225
12  CoreFoundation                       0x00007fff2b3818d0 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1664
13  CoreFoundation                       0x00007fff2b380a07 _CFXNotificationPost + 599
14  Foundation                           0x00007fff2d4908c7 -[NSNotificationCenter postNotificationName:object:userInfo:] + 65
15  AppKit                               0x00007fff28a69973 -[NSWindow becomeKeyWindow] + 1487
16  AppKit                               0x00007fff28a68bcd -[NSWindow _changeKeyAndMainLimitedOK:] + 877
17  AppKit                               0x00007fff28ae3ca8 -[NSWindow _orderOutAndCalcKeyWithCounter:stillVisible:docWindow:] + 1260
18  AppKit                               0x00007fff28a6bc0c NSPerformVisuallyAtomicChange + 145
19  AppKit                               0x00007fff29219833 -[NSWindow _doWindowOrderOutWithWithKeyCalc:forCounter:orderingDone:docWindow:] + 84
20  AppKit                               0x00007fff2921ad26 -[NSWindow _reallyDoOrderWindowOutRelativeTo:findKey:forCounter:force:isModal:] + 445
21  AppKit                               0x00007fff28a738ec -[NSWindow _reallyDoOrderWindow:relativeTo:findKey:forCounter:force:isModal:] + 171
22  AppKit                               0x00007fff28a72765 -[NSWindow _doOrderWindow:relativeTo:findKey:forCounter:force:isModal:] + 286
23  AppKit                               0x00007fff28a725cb -[NSWindow orderWindow:relativeTo:] + 168
24  AppKit                               0x00007fff2921840e -[NSWindow _finishClosingWindow] + 478
25  AppKit                               0x00007fff28bb7ed8 -[NSWindow _close] + 377
26  Sparkle                              0x000000010aae8d57 0x10aab9000 + 195927
27  Sparkle                              0x000000010aae91fe 0x10aab9000 + 197118
28  Sparkle                              0x000000010aabbee1 0x10aab9000 + 12001
29  Sparkle                              0x000000010aad571f 0x10aab9000 + 116511
30  Sparkle                              0x000000010aac9a06 0x10aab9000 + 68102
31  Sparkle                              0x000000010aac9271 0x10aab9000 + 66161
32  CFNetwork                            0x00007fff2a497088 __82-[NSURLSession delegate_downloadTask:didFinishDownloadingToURL:completionHandler:]_block_invoke + 37
33  Foundation                           0x00007fff2d4c65df __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 6
34  Foundation                           0x00007fff2d4c6441 -[NSBlockOperation main] + 67
35  Foundation                           0x00007fff2d4c48ee -[__NSOperationInternal _start:] + 777
36  libdispatch.dylib                    0x00007fff53182db8 _dispatch_client_callout + 7
37  libdispatch.dylib                    0x00007fff5319533e _dispatch_block_invoke_direct + 316
38  libdispatch.dylib                    0x00007fff53182db8 _dispatch_client_callout + 7
39  libdispatch.dylib                    0x00007fff5319533e _dispatch_block_invoke_direct + 316
40  libdispatch.dylib                    0x00007fff531951e1 dispatch_block_perform + 108
41  Foundation                           0x00007fff2d4c08fc __NSOQSchedule_f + 341
42  libdispatch.dylib                    0x00007fff53182db8 _dispatch_client_callout + 7
43  libdispatch.dylib                    0x00007fff53195e81 _dispatch_continuation_pop + 471
44  libdispatch.dylib                    0x00007fff5318d733 _dispatch_async_redirect_invoke + 702
45  libdispatch.dylib                    0x00007fff531849a9 _dispatch_root_queue_drain + 514
46  libdispatch.dylib                    0x00007fff53184755 _dispatch_worker_thread3 + 100
47  libsystem_pthread.dylib              0x00007fff534d4169 _pthread_wqthread + 1386
48  libsystem_pthread.dylib              0x00007fff534d3be9 start_wqthread + 12
```

**Comment by kornelski** [comment]
This looks like a bug that was fixed recently. Please try the latest beta: https://github.com/sparkle-project/Sparkle/releases/tag/1.20.0b1

**Comment by Kapeli** [comment]
Confirmed fixed in 1.20.0.
